edb/server/server.py

```diff
@@ -934,18 +944,20 @@ async def _introspect_dbs(self):
                 # information about them.
                 g.create_task(self._early_introspect_db(dbname))
 
-    async def get_patch_count(self, conn):
+    async def get_patch_count(self, conn: pgcon.PGConnection) -> int:
         """Get the number of applied patches."""
-        num_patches = await conn.sql_fetch_val(
+        num_patches_b = await conn.sql_fetch_val(
             b'''
                 SELECT json::json from edgedbinstdata.instdata
                 WHERE key = 'num_patches';
             ''',
         )
-        num_patches = json.loads(num_patches) if num_patches else 0
+        num_patches = json.loads(num_patches_b) if num_patches_b else 0
         return num_patches
 
-    async def _get_patch_log(self, conn, idx):
+    async def _get_patch_log(
+        self, conn: pgcon.PGConnection, idx: int
+    ) -> Optional[bootstrap.PatchEntry]:
```

Comment:
We prefer `Optional[T]` over `T | None`, is that right?

Reply:
*I* prefer `Optional[T]`, at least, and most of the code base predates being able to write `T | None` anyway